### PR TITLE
:book: Improve documentation.

### PR DESCRIPTION
**Pull request by guettli**
**What this PR does / why we need it**:

Improve the documentation.

Here is the rendered preview of [README.md](https://github.com/syself/cluster-api-provider-hetzner/blob/tg/improve-docs/README.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,12 +52,33 @@ Using the Cluster API Provider Hetzner (CAPH) unites the benefits of declarative
...
 * Complete day 2 operations - updating Kubernetes and nodes, scaling up and down, self-healing
 * Custom CSR approver for approving [kubelet-serving certificate signing requests](https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/kubeadm-certs/#kubelet-serving-certs)
-* Hetzner dedicated servers / bare metal
+* Hetzner dedicated servers / bare metal (and GPUs)
+
+## :eyes: Clarifying Scope
 
+Managing a production-grade Kubernetes system requires a **seasoned team of experts**.
 
+The Cluster API Provider Hetzner (CAPH) plays a specific, limited role within this intricate ecosystem. It deals with the low-level lifecycle management of machines and infrastructure. While it can support the creation of production-ready Kubernetes clusters, it does not define the specific contents within the cluster. This important task is something that needs to be handled independently.
```
Comment (janiskemper) [inline README.md:61]: the specific content?

**Review by janiskemper — CHANGES_REQUESTED** [review]
I didn't understand this one sentence. Otherwise good

File: README.md [hunk 2]
```diff
@@ -22,7 +22,7 @@
 The Kubernetes Cluster API Provider Hetzner enables declarative provisioning of multiple Kubernetes clusters on [Hetzner infrastructure](https://hetzner.cloud).
 High-available Kubernetes clusters on bare metal and on cloud instances are supported. The Cluster API simplifies the creation, updates, and operation of production-ready clusters. Self-managed Kubernetes clusters can, therefore, be easily operated even at a large scale.
 
-> Disclaimer: This is no official Hetzner project! It is maintained by the folks of the cloud-native start-up [Syself](https://syself.com).
+> Disclaimer: This is no official Hetzner project! It is maintained by [Syself](https://syself.com).
```
Comment (buhodev) [inline README.md:25]: Maybe would be better to say "This is not an official..." ?

Comment (guettli) [reply README.md:25]: @buhodev yes, thank you "... not an ..." is better.

File: README.md (hunk 1 above)
Comment (guettli) [reply README.md:62]: @batistein about ".. it does not define the specific contents within the cluster." This wording was from a different PR. Can you elaborate?

**Review by batistein — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (janiskemper) [reply README.md]: I just don't understand what is meant by it... Maybe we can change it. But probably doesn't block this

**Pull request merged**